streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -62,7 +67,11 @@ public void init(final ProcessorContext context) {
             super.init(context);
             metrics = (StreamsMetricsImpl) context.metrics();
 
-            otherWindow = (WindowStore<K, V2>) context.getStateStore(otherWindowName);
+            StateStore store = context.getStateStore(otherWindowName);
```

Comment:
Hmm.. we are pealing off two layers here. The second layer is window-store-facade, what's the first layer of WrappedStateStore?

Ditto elsewhere.

Reply:
For some stores, it sometimes one layer, sometimes multiple. Thus, this is a guard to get the right store for any case -- I use the same code for all DSL processor for simplicity in this PR. Might not be necessary for all processors.